modules/@apostrophecms/schema/ui/apos/components/AposInputAttachment.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <AposInputWrapper
+    :field="field" :error="error"
+    :uid="uid"
+  >
+    <template #body>
+      <div class="apos-attachment">
+        <label
+          class="apos-input-wrapper apos-attachment-dropzone"
+          :class="{ 'apos-attachment-dropzone--dragover': dragging }"
+          :disabled="disabled"
+          @drop.prevent="uploadMedia"
+          @dragover="dragHandler"
+          @dragleave="dragging = false"
+        >
+          <p class="apos-attachment-instructions">
+            <template v-if="dragging">
+              <cloud-upload-icon :size="38"  />
+            </template>
+            <template v-else>
+              <paperclip-icon :size="14" class="apos-attachment-icon"/>
+              {{ messagePart1 }}&nbsp;
+              <span class="apos-attachment-highlight">{{ messagePart2 }}</span>
+            </template>
+          </p>
+          <input
+            type="file"
+            class="apos-sr-only"
+            :disabled="disabled"
+            @input="uploadMedia"
+          >
+        </label>
+        <div v-if="next" class="apos-attachment-files">
+          <AposSlatList
+            :initial-items="[ next ]"
+            @update="updated"
+          />
+        </div>
+      </div>
+    </template>
+  </AposInputWrapper>
+</template>
+
+<script>
+import AposInputMixin from 'Modules/@apostrophecms/schema/mixins/AposInputMixin.js';
+import AposSlatList from 'Modules/@apostrophecms/ui/components/AposSlatList';
+
+export default {
+  name: 'AposInputAttachment',
+  components: {
+    AposSlatList
+  },
+  mixins: [ AposInputMixin ],
+  emits: [ 'upload-started', 'upload-complete' ],
+  data () {
+    return {
+      disabled: undefined,
+      dragging: false,
+      messagePart1: '',
+      messagePart2: ''
+    };
+  },
+  watch: {
+    disabled(value) {
+      if (value) {
+        this.messagePart1 = 'This field is disabled';
+        this.messagePart2 = '';
+      } else {
+        this.messagePart1 = 'Drop a file here or';
+        this.messagePart2 = 'click to open the file explorer';
+      }
+    }
+  },
+  async mounted () {
+    this.disabled = this.field.disabled || !!(this.value.data && this.value.data._id);
+  },
+  methods: {
+    watchNext () {
+      this.validateAndEmit();
+      this.disabled = !!this.next.length;
```

Comment:
If an existing attachment is removed by the user, `this.next` becomes an empty array. In this case, `disabled` becomes `false`.